src/Rings/mpolyquo-localizations.jl

```diff
@@ -2565,7 +2565,41 @@ function small_generating_set(
   end::Vector{elem_type(base_ring(I))} 
 end
 
-dim(R::MPolyQuoLocRing{<:Field, <:FieldElem, <:MPolyRing, <:MPolyRingElem, <:MPolyComplementOfPrimeIdeal}) = dim(saturated_ideal(modulus(R))) - dim(prime_ideal(inverted_set(R)))
+@attr Int function dim(R::MPolyQuoLocRing)
+  error("Not implemented")
+end
+
+@attr Int function dim(R::MPolyLocRing)
+  error("Not implemented")
+end
+
+@attr Int function dim(R::MPolyQuoLocRing{<:Any, <:Any, <:MPolyRing, <:MPolyRingElem, <:Union{MPolyComplementOfPrimeIdeal, MPolyComplementOfKPointIdeal}})
+  P = prime_ideal(inverted_set(R))
+  I = saturated_ideal(modulus(R))
+  return dim(I) - dim(P)
+end
+
+@attr Int function dim(R::MPolyQuoLocRing{<:Any, <:Any, <:MPolyRing, <:MPolyRingElem, <:MPolyPowersOfElement})
+  return dim(closure(spec(R)))
+end
+
+@attr Int function dim(R::MPolyLocRing{<:Any,<:Any,<:MPolyRing,<:MPolyRingElem, <:MPolyPowersOfElement})
+  # zariski open subset of A^n
+  return dim(closure(spec(R)))
+end
```

Comment:
You are again passing through the scheme side here. Please try to use saturated ideal instead of closure. It should also have a variant for MPolyPowersOfElement.